lightkurve/targetpixelfile.py

```diff
@@ -41,6 +44,110 @@ def __init__(self, path, quality_bitmask='default', targetid=None, **kwargs):
+            DEC: Declination [degrees]    (KIC & EPIC)
+            RA: Right ascension [degrees] (Gaia)
+            DEC: Declination [degrees] (Gaia)
+            pmRA: Proper motion for right ascension [mas/year]
+            pmDEC: Proper motion for declination [mas/year]
+            Kpmag: Magnitude in Kepler band [mag] (KIC & EPIC)
+            Gmag: Magnitude in Gaia band [mag]  (Gaia)
+            column: Pixel column coordinate [pix]
+            row: Pixel row coordinate [pix]
+            predicted_flux: Predicted flux of source [e-/s]
+            epoch: Epoch of data [years]
+
+        """
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+                catalog_epoch = 2000.0
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+                catalog_epoch = 2000.0
+            else:
+                raise ValueError('Please provide a catalog. Catalog must be KIC, EPIC or Gaia.')
+        else:
+            catalog_epoch = 2015.5
+
+        if dist_tolerance < 3:
+            log.warning('Distance tolerance is less than 3 pixels. Sources near the edge will likely be missed.')
+
+        #Skycoord the centre of target
+        cent = SkyCoord(ra=self.ra, dec=self.dec, frame='icrs', unit=(u.deg, u.deg))
+
+        #Find the size of the TPF
+        radius = ((np.max(self.flux.shape[1:2]) * 4) + dist_tolerance)/ 60.0  # arcmin
+
+        # query around centre with radius and catalog
+        tbl = query_catalog(cent, radius=radius, catalog=catalog)
+
+        # Find where nans are in cadence
+        tpf_mask = np.any(np.isfinite(self.flux), axis=0)
+
+        # Coordinates of sky sources and tpf pixels
+        pixels_ra, pixels_dec = self.get_coordinates(cadence=int(len(self.cadenceno)/2))
+        pixels_radec = np.asarray([pixels_ra[tpf_mask].ravel(), pixels_dec[tpf_mask].ravel()])
+        sky_pixel_pairs = SkyCoord(ra=pixels_radec[0], dec=pixels_radec[1], frame='icrs', unit=(u.deg, u.deg))
+        sky_sources = SkyCoord(ra=tbl['ra'], dec=tbl['dec'], frame='icrs', unit=(u.deg, u.deg))
+
+        separation_mask = np.zeros(len(tbl), dtype=bool)
+        for idx in range (len(tbl)):
+            sep = sky_pixel_pairs.separation(sky_sources[idx])  # Estimate separation
+            separation = np.any(sep.arcsec <= dist_tolerance)
+            separation_mask[idx] = separation
+        sep_mask = np.array(separation_mask, dtype=bool)
+
+        # constrain with magnitude limit
+        sep_mask &= tbl['mag'] < magnitude_limit
+
+        # After applying filters we can now correct the posistions
+        source_list = tbl[sep_mask]
+
+        # Transform coordinates to median epoch of data
+        tpf_epoch = np.median(self.astropy_time.byear)
+        has_pm = ~(source_list['pmra'].mask | source_list['pmdec'].mask)
+        source_list['ra'][has_pm] += (tpf_epoch - catalog_epoch) * (source_list['pmra'][has_pm] * (2.778e-7)) # mas to degree
```

Comment:
Astropy already has a nice way of dealing with units, are you sure the units of these catalogs are always in mas/year? If yes I have no problem, but in future it might be worth using the inbuilt astropy units feature rather than hard coding a conversion.